home-screen-widget/profiles/DEVELOPERS_PORTAL.md

```diff
@@ -0,0 +1,412 @@
+
+**Registering a widget**
+
+When we register a widget we make the widget component known to the home screen. The widget component is the Vue.js component and the widgetTypeId is a unique identifier for the widget (e.g. the Streams widget has id 'streams').
+
+```javascript
+registerWidget(widgetTypeId: string, widgetComponent: Component): void;
+```
+
+example: 
+
+```javascript
+import BeekeeperHomeScreen from '@beekeeper/home-screen-sdk';
+
+import component, { WIDGET_ID } from '~/profiles/components/Widget.vue';
+BeekeeperHomeScreen.registerWidget(WIDGET_ID, component)
+```
+
+
+**Sending events to the home screen**
+
+Widgets can communicate to the home screen by triggering events. 
+
+```javascript 
+  triggerEvent(
+    eventName: EventType,
+    widgetInstanceId: string,
+    payload?: WidgetEventPayload
+  ): void;
+
+```
+ 
+
+For example this is how we trigger the LOADED event in the [example widget](https://github.com/beekpr/examples/tree/master/home-screen-widget/profiles).
+```javascript 
+  import BeekeeperHomeScreen, { EventType } from '@beekeeper/home-screen-sdk';
+  BeekeeperHomeScreen.triggerEvent(EventType.LOADED, this.widgetInstanceId);
+```
+
+**Calling the Javascript SDK API**
+
+The [Javascript SDK API](https://developers.beekeeper.io/v2/welcome/javascript-sdk) gives access to tenant data such as messages, profiles, streams. 
+
+Example from the [profiles widget](https://github.com/beekpr/examples/tree/master/home-screen-widget/profiles):
+
+```javascript 
+  import BeekeeperHomeScreen from '@beekeeper/home-screen-sdk';
+  ...
+  BeekeeperHomeScreen.sdk.Profiles.list({ limit: 50 });
+```
+
+**Accessing the mobile bridge**
+ 
+The [mobile bridge](https://developers.beekeeper.io/v2/welcome/mobile-bridge) enables the interaction of a hybrid feature with the native host application (i.e., the Android or iOS app) via the Beekeeper Bridge.
+
+Example from the [profiles widget](https://github.com/beekpr/examples/tree/master/home-screen-widget/profiles):
+
+```javascript 
+  import BeekeeperHomeScreen from '@beekeeper/home-screen-sdk';
+  ...
+  const BeekeeperMobileBridge = BeekeeperHomeScreen.bridge;
+  initI18n(BeekeeperMobileBridge.ui.locale)
+```
+Here we fetch the device locale in order to initialize the translation library. 
+
+## 2. Home screen widget development
+
+### 2.1. Set up local development environment 
+TODO
+
+### 2.2 Developing a new widget
+
+In order to facilitate the development of third-party widgets we created an example widget. 
+The widget uses the Beekeeper SDK accessible through the HomeScreenSDK to fetch and display's user profiles. 
+
+Make sure to follow the instruction steps in the code to find the essential bits and pieces. 
+
+The code can be found [here](https://github.com/beekpr/examples/tree/FUL-21524/home-screen-widget/profiles).
+
+
+### 2.3 Publishing a widget
+
+TODO
+
+
+## 3. Configuring the home screen for a tenant
+The home screen and the widgets can be configured by the tenants.\
```

Comment:
What does this mean?

Reply:
I changed to "The home screen and the widgets can be configured" and right below I elaborate a bit more